File: reference-app/src/main/java/org/smartregister/sample/fp/job/ViewConfigurationsServiceJob.java
package org.smartregister.sample.fp.job;

import android.content.Intent;

import androidx.annotation.NonNull;

import com.evernote.android.job.Job;

import org.smartregister.configurableviews.service.PullConfigurableViewsIntentService;
import org.smartregister.fp.common.util.ConstantsUtils;
import org.smartregister.job.BaseJob;

/**
 * Created by ndegwamartin on 06/09/2018.
 */
public class ViewConfigurationsServiceJob extends BaseJob {

    public static final String TAG = "ViewConfigurationsServiceJob";

    @NonNull
    @Override
    protected Job.Result onRunJob(@NonNull Job.Params params) {
        Intent intent = new Intent(getApplicationContext(), PullConfigurableViewsIntentService.class);
        getApplicationContext().startService(intent);
        return params != null && params.getExtras().getBoolean(ConstantsUtils.IntentKeyUtils.TO_RESCHEDULE, false) ?
                Result.RESCHEDULE : Result.SUCCESS;
    }
}

File: reference-app/src/main/java/org/smartregister/sample/fp/login/job/ViewConfigurationsServiceJob.kt
package org.smartregister.sample.fp.job

import android.content.Intent
import androidx.annotation.NonNull
import org.smartregister.configurableviews.service.PullConfigurableViewsIntentService
import org.smartregister.fp.common.util.ConstantsUtils
import org.smartregister.job.BaseJob

class ViewConfigurationsServiceJob : BaseJob() {

    fun fetchParms(): Params {
        return params
    }

    @NonNull
    override fun onRunJob(@NonNull params: Params): Result {
        val intent = Intent(applicationContext, PullConfigurableViewsIntentService::class.java)
        applicationContext.startService(intent)
        return if (params != null && params.extras.getBoolean(ConstantsUtils.IntentKeyUtils.TO_RESCHEDULE, false)) Result.RESCHEDULE else Result.SUCCESS
    }

    companion object {
        const val TAG = "ViewConfigurationsServiceJob"
    }

}